opcua.cpp

```diff
@@ -780,6 +929,23 @@ int OPCUA::subscribe()
 	int i = 0;
 	for (auto it = variables.cbegin(); it != variables.cend(); it++)
 	{
+		std::string browseName;
+		OpcUa_NodeClass nodeClass;
+		bool rv = Node::getNodeAttr(m_connectionId, *it, browseName, nodeClass);
+
+		bool processNode = true;  // include child node by default if they are variables
+		
+		// If parent node is being subscribed to, children variable nodes are also subscribed to
+		// And, if the child node is an object, it is evaluated independently as per configured filter
+		if(nodeClass != OpcUa_NodeClass_Variable)
+			processNode = checkNode(*it, false);
+		
+		if(!processNode)
+		{
+			logger->warn("Skipping subscription for node '%s' because of filtering config", it->c_str());
+			continue;
+		}
+		
 		try
 		{
 			// Hack: Skip any NodeId string with leading underscores in any segment of the path.
```

Comment:
Lines 951 through 957: this was a temporary hack that needs to be removed. Your regex filtering now does this job.